Improve accessibility of company info expansion panel by adding keyboard focus handling and visual indicators

kivuvarosekivuvan opened the pull request:
## Description
This PR addresses the issue with keyboard-inaccessibility of the company-info section. 

Fixes: https://github.com/ushahidi/platform/issues/4961

This is a change from the merged PR that addressed the same: https://github.com/ushahidi/platform-client-mzima/pull/1127

kivuvarosekivuvan committed:
Improve accessibility of company info expansion panel by adding keyboard focus handling and visual indicators

kivuvarosekivuvan committed:
Improve accessibility of company info expansion panel by adding keyboard focus handling and visual indicators

kivuvarosekivuvan commented:
### @Ifycode Hey 👋,
I've found a couple of things while working on this:

**1. Focus Issue with Open Accordion:** With the current changes, the focus appears to be on the header once the accordion is open. This means the focus outline is still not visible. I've tried conditionally applying CSS styles to show or hide the focus outline based on the input method using ngClass but still the same issue.

**2. Alternative Approach with focus-within:** Alternatively, we can achieve the desired behavior by directly applying the focus styles on the .company-info element using the :focus-within pseudo-class. This approach seems to work, but it may require some minor adjustments to the styling to get the desired look and feel. I have also tried this approach

```
.company-info {
  &:focus-within {
    outline: 1px solid #ffffff;
  }
}
```

The second approach using **:focus-within** seems like a more straightforward solution, since it'll allow us to apply the focus styles directly on the element. However, I really think this will need us to fine-tune the styling to achieve what we have on the other focus rings.

Please let me know what you think of this issue

Ifycode commented on apps/web-mzima-client/src/app/shared/components/company-info/company-info.component.scss:
```diff
@@ -113,6 +113,16 @@
+.has-outline:focus,
+.has-outline:hover {
+  outline: 1px solid rgb(255, 255, 255);
+}
```
@kivuvarosekivuvan based on the findings you wrote in our slack chat (and now that I think about it more), we don't need a new class-name in the css. We also don't need all those javascript code. You can happily remove all the code you have added.

I've also seen your comment here on using focus-within inside of the company-info class. I think it may be more helpful and clear to developers if we target the class from the library/framework directly.

I'll suggest a more straightforward fix using only css. At the bottom of this `company-info.component.scss` file, you will find the `.mat-expansion-panel` class.

**_Step 1:_** You can restore/fix the **_focus_** of the `.mat-expansion-pane-header` from within here.

Now, after doing this, most parts of the focus ring on the header is likely to be hidden. This is because `mat-expansion-panel` element itself has some hidden overflow applied to it.

**_Step 2:_** You can also just restore/fix the overflow of the `mat-expansion-panel` element from within there also.

Really, that's all.

Ifycode reviewed (changes requested):
Hmm...

kivuvarosekivuvan committed:
Restore focus and overflow for mat-expansion-panel

This commit addresses two issues with the mat-expansion-panel component:

1. Restored the focus ring on the mat-expansion-panel-header when the panel is focused. Previously, the focus ring was being hidden.

2. Restored the overflow property on the mat-expansion-panel element to 'visible'. This ensures that the content within the panel is not hidden due to any overflow settings.

kivuvarosekivuvan committed:
Restore focus and overflow for mat-expansion-panel

This commit addresses two issues with the mat-expansion-panel component:

1. Restored the focus ring on the mat-expansion-panel-header when the panel is focused. Previously, the focus ring was being hidden.

2. Restored the overflow property on the mat-expansion-panel element to 'visible'. This ensures that the content within the panel is not hidden due to any overflow settings.

kivuvarosekivuvan committed:
Restore focus and overflow for mat-expansion-panel

This commit addresses two issues with the mat-expansion-panel component:

1. Restored the focus ring on the mat-expansion-panel-header when the panel is focused. Previously, the focus ring was being hidden.

2. Restored the overflow property on the mat-expansion-panel element to 'visible'. This ensures that the content within the panel is not hidden due to any overflow settings.

kivuvarosekivuvan committed:
Restore focus and overflow for mat-expansion-panel

This commit addresses two issues with the mat-expansion-panel component:

1. Restored the focus ring on the mat-expansion-panel-header when the panel is focused. Previously, the focus ring was being hidden.

2. Restored the overflow property on the mat-expansion-panel element to 'visible'. This ensures that the content within the panel is not hidden due to any overflow settings.

Ifycode commented on apps/web-mzima-client/src/app/shared/components/company-info/company-info.component.scss:
```diff
@@ -116,9 +116,13 @@
     .mat-expansion-panel-header {
+      &:focus {
+        outline: 1px solid #fdfdfd;
+      }
```
Because it's inside `&:not(.mat-expanded):not([aria-disabled='true'])`, it would only work after the dropdown is expanded or so.
You can move it outside of it @kivuvarosekivuvan

Ifycode replied:
Also let's use the browser default focus outline instead for consistency.

Ifycode commented on apps/web-mzima-client/src/app/shared/components/company-info/company-info.component.scss:
```diff
@@ -116,9 +116,13 @@
       }
     }
   }
+  overflow: visible;
```
Move this to the top of this code block for easy readability @kivuvarosekivuvan

kivuvarosekivuvan committed:
Updated the outline to the default one the browser uses

kivuvarosekivuvan committed:
Updated the outline to the default one the browser uses

kivuvarosekivuvan commented:
Hoyee 🤩, @Ifycode Changes upto date 
cc: @Angamanga

Ifycode commented on apps/web-mzima-client/src/app/shared/components/company-info/company-info.component.scss:
```diff
@@ -114,6 +114,14 @@
+
+  .mat-expansion-panel-header {
+    &:focus {
+      outline: auto;
```
Nice. Try `revert !important` instead of `auto` @kivuvarosekivuvan

kivuvarosekivuvan committed:
Ensure radio button focus is always visible by adding important modifier

kivuvarosekivuvan committed:
Ensure radio button focus is always visible

kivuvarosekivuvan committed:
Merge remote changes with local changes

Ifycode committed:
Merge branch 'development' into company-focus

Angamanga committed:
Merge branch 'development' into company-focus

Angamanga committed:
Merge branch 'development' into company-focus
